### PR TITLE
Use (and pass) Clang's `-Wbad-function-cast` check and enable in CI

### PR DESCRIPTION
**Pull request by johnsonjh**
* Use (and pass) Clang's `-Wbad-function-cast` check and enable in CI

### CODE REVIEW

**Review by aremmell — APPROVED** [review]

**Comment by aremmell** [comment]
I forgot about the LAN address in the script... need to fix that before we pres butan 2 merge

**Review by aremmell — CHANGES_REQUESTED** [review]
Just need to get rid of the LAN IP in Dockerfile...

File: Makefile [hunk 1]
```diff
@@ -175,7 +175,7 @@ $(OUT_TESTS): $(OUT_STATIC) $(OBJ_TESTS)
 	@mkdir -p $(BINDIR)
 	@mkdir -p $(LOGDIR)
 	@touch $(BINDIR)/file.exists > /dev/null
-	$(CC) -o $(OUT_TESTS) $(OBJ_TESTS) -Iinclude $(LIBSIR_S) $(LDFLAGS)
+	$(CC) -o $(OUT_TESTS) $(OBJ_TESTS) -Iinclude $(LIBSIR_S) $(LDFLAGS) -lm
```
[the remark this answers is not in the document]
Comment (johnsonjh) [reply Makefile:178]: It's for lroundf - and we don't want to link the math library for everything - only for sirtests

[the remark this answers is not in the document]
Comment (johnsonjh) [reply Makefile:178]: Also, it's only needed for Unix - it's just in libc in windows, so no changes needed.

**Comment by johnsonjh** [comment]
> Just need to get rid of the LAN IP in Dockerfile...

@aremmell It's correct, but I added an argument so you can override locally without editing via `--build-arg URL="..."`

**Comment by aremmell** [comment]
Erm, isn't the pluginexport just a typedef void*?

**Comment by johnsonjh** [comment]
> Erm, isn't the pluginexport just a typedef void*?

Seems to be. MSVC doesn't like it.

**Comment by aremmell** [comment]
> > Erm, isn't the pluginexport just a typedef void*?
> 
> Seems to be. MSVC doesn't like it.

It's because on Windows, it's typedef FARPROC.

**Comment by johnsonjh** [comment]
> > > Erm, isn't the pluginexport just a typedef void*?
> > 
> > 
> > Seems to be. MSVC doesn't like it.
> 
> It's because on Windows, it's typedef FARPROC.

Well Windows sucks for being weird.

**Review by aremmell — APPROVED** [review]

**Pull request merged**